plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py

```diff
@@ -0,0 +1,127 @@
+        "temperature_of_melting": to_kelvin(Celsius(1_083.4)),
+    },
+    "Au": {
+        "specific_heat_heating": 129,
+        "specific_heat_melting": 67_000,
+        "temperature_of_melting": to_kelvin(Celsius(1_063)),
+    },
+    "Zn": {
+        "specific_heat_heating": 400,
+        "specific_heat_melting": 117_000,
+        "temperature_of_melting": to_kelvin(Celsius(415)),
+    }
+}
+
+velocity_of_meteorite = symbols("velocity_of_meteorite")
+temperature_of_meteorite = symbols("temperature_of_meteorite")
+koefficient_of_transfer_to_kinetic = symbols("koefficient_of_transfer_to_kinetic")
+
+specific_heat_heating_meteorite = symbols("specific_heat_heating_meteorite")
+specific_heat_melting_meteorite = symbols("specific_heat_melting_meteorite")
+temperature_of_meteorite_melting = symbols("temperature_of_meteorite_melting")
+
+mass_of_meteorite = symbols("mass_of_meteorite")
+mass_of_melting_meteorite = symbols("mass_of_melting_meteorite")
+koefficient_of_melting_meteorite = symbols("koefficient_of_melting_meteorite")
+
+energy_to_heating_meteorite = energy_heating_law.law.subs({
+    energy_heating_law.specific_heat_capacity: specific_heat_heating_meteorite,
+    energy_heating_law.body_mass: mass_of_meteorite,
+    energy_heating_law.temperature_origin: temperature_of_meteorite,
+    energy_heating_law.temperature_end: temperature_of_meteorite_melting
+}).rhs
+
+energy_to_meteorite_melting = energy_melting_law.law.subs({
+    energy_melting_law.mass_of_matter: mass_of_melting_meteorite,
+    energy_melting_law.specific_heat_melting: specific_heat_melting_meteorite
+}).rhs
+
+kinetic_energy_of_meteorite = kinetic_energy_law.law.subs({
+    kinetic_energy_law.body_mass: mass_of_meteorite,
+    kinetic_energy_law.body_velocity: velocity_of_meteorite
+}).rhs
+
+energy_conservation_equation = energy_conservation_law.law.subs({
+    energy_conservation_law.mechanical_energy(energy_conservation_law.time_before): koefficient_of_transfer_to_kinetic * kinetic_energy_of_meteorite,
+    energy_conservation_law.mechanical_energy(energy_conservation_law.time_after): energy_to_heating_meteorite + energy_to_meteorite_melting
+})
+mass_of_melting_value = solve(energy_conservation_equation, mass_of_melting_meteorite,
+    dict=True)[0][mass_of_melting_meteorite]
+koefficient_of_melting_meteorite_value = mass_of_melting_value / mass_of_meteorite
+answer = Eq(koefficient_of_melting_meteorite, koefficient_of_melting_meteorite_value)
+print(f"Total equation:\n{print_expression(answer)}")
+
+koefficient_of_melting_meteorite_to_plots = answer.subs({
+    koefficient_of_transfer_to_kinetic: 0.80,   # percents
+    temperature_of_meteorite: 300   # kelvins
+})
+
+base_plot = plot(title="The proportion of a molten meteorite depending on its velocity",
+    xlabel=r"$v, m/s$",
+    ylabel=r"$m_{melt-meteorite} / m_{meteorite}$",
+    backend=MatplotlibBackend,
+    legend=True,
+    show=False)
+
+masses_ratio_supremum = 1
+masses_ratio_minimum = 0
```

Comment:
There is convention to write constants in all caps, eg `MASSES_RATIO_SUPREMUM`
BTW why supremum, not `maximum`?